### ISSUE TITLE
Hub And Spoke build fails due to error resolving image name

### ISSUE THREAD

**Issue by daniel-cit** [terraform-google-modules/terraform-example-foundation#600]
### TL;DR

The Hub And Spoke build is failing  due to an error resolving the image name ` centos-cloud/cos-stable-85-13310-1041-161`

### Expected behavior

Build `Integration-tests-for-examplefoundations-in-HubAndSpoke-mode` should be successful. 

### Observed behavior

Build Fails with this error:

```
Error: Error resolving image name 'centos-cloud/cos-stable-85-13310-1041-161': Could not find image or family centos-cloud/cos-stable-85-13310-1041-161

  on .terraform/modules/shared.base_transitivity.templates/modules/instance_template/main.tf line 58, in resource "google_compute_instance_template" "tpl":
  58: resource "google_compute_instance_template" "tpl" {
``` 

### Terraform Configuration

```hcl
Any build that runs the `Integration-tests-for-examplefoundations-in-HubAndSpoke-mode` trigger
```

### Terraform Version

```sh
build uses 
gcr.io/cloud-foundation-cicd/cft/developer-tools:0.13
```

### Additional information

- The location in the code for the image definition

https://github.com/terraform-google-modules/terraform-example-foundation/blob/0ee39ceff2dce8c3a3b61d59bb1d45a4d0687e74/3-networks/modules/transitivity/main.tf#L49

- A failing build => https://github.com/terraform-google-modules/terraform-example-foundation/pull/599/checks?check_run_id=4455890232

**Comment by daniel-cit** [comment]
Last  `Integration-tests-for-examplefoundations-in-HubAndSpoke-mode` build https://console.cloud.google.com/cloud-build/builds/2c575386-ebc9-494d-9a81-c3b7c0e21e5d?project=2843445864 did not fail

**Comment by bharathkkb** [comment]
I would have expected `Could not find image or family` if this image was removed/deprecated, odd that it came back.

**Comment by dhess** [comment]
This is happening for me now on `3-networks` in my new deployment.

**Comment by dhess** [comment]
Adding `source_image_project = "cos-cloud"` fixed the issue for me. This appears to be the correct family anyway:

```sh
$ gcloud compute images list --project="cos-cloud" 
NAME                                                  PROJECT              FAMILY                            DEPRECATED  STATUS
centos-7-v20211214                                    centos-cloud         centos-7                                      READY
centos-8-v20211214                                    centos-cloud         centos-8                                      READY
centos-stream-8-v20211214                             centos-cloud         centos-stream-8                               READY
cos-81-12871-1317-7                                   cos-cloud            cos-81-lts                                    READY
cos-85-13310-1366-14                                  cos-cloud            cos-85-lts                                    READY
cos-89-16108-534-34                                   cos-cloud            cos-89-lts                                    READY
cos-93-16623-39-30                                    cos-cloud            cos-93-lts                                    READY
cos-beta-93-16623-39-6                                cos-cloud            cos-beta                                      READY
debian-10-buster-v20211209                            debian-cloud         debian-10                                     READY
...
```
